### ISSUE TITLE
Remove deprecated t.p.reflect.macro()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5035]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5035|
|Type|task|
|Created|2011-04-07 16:56:34Z|
|Branch|https://github.com/twisted/twisted/tree/reflect-macro-5035|

[twisted/python/reflect.py](twisted/python/reflect.py) has a `macro` method that was deprecated in Twisted 8.2, should be safe to remove it now.

<details><summary>Searchable metadata</summary>

```
trac-id__5035 5035
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_reflect_macro_5035 branches/reflect-macro-5035
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1302195394000000 1302195394000000
changetime__1331150122000000 1331150122000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [33565]) Branching to 'reflect-macro-5035'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [33566]) Remove `macro`, adjust docstrings to coding standard. refs [#5035](https://github.com/twisted/twisted/issues/5035)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set owner to @thijstriemstra|
|-|-|

In the future, you could start a buildbot run on your changes using `force-builds.py` and include a link in the ticket?

Assuming all tests pass, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [itamar](#note_6):
> In the future, you could start a buildbot run on your changes using `force-builds.py` and include a link in the ticket?
> 
> Assuming all tests pass, please merge.

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/reflect-macro-5035).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33661]) Merge reflect-macro-5035: Removed twisted.python.reflect.macro, deprecated since Twisted 8.2.

Author: thijs
Reviewer: itamar
Fixes: [#5035](https://github.com/twisted/twisted/issues/5035)
